### ISSUE TITLE
[production][binaries#show] No route matches {:action=>"index", :arch=>"repo", :controller=>"webui/packages/binaries", :filen...

### ISSUE THREAD

**Issue by saraycp** [openSUSE/open-build-service#14879]
[See this exception on Errbit](https://errbit.opensuse.org/apps/5620ca2bdc71fa00b1000000/problems/6509967884bf460703acd9df)
# No route matches {:action=&gt;&quot;index&quot;, :arch=&gt;&quot;repo&quot;, :controller=&gt;&quot;webui/packages/binaries&quot;, :filename=&gt;&quot;repoindex.xml&quot;, :package_name=&gt;&quot;telegram-desktop&quot;, :project_name=&gt;&quot;openSUSE:Factory&quot;, :repository_name=&gt;nil}, possible unmatched constraints: [:repository_name]
Did you mean?  project_package_repository_binaries_url #
### URL ###
[https://build.opensuse.org/projects/openSUSE:Factory/packages/telegram-desktop/repositories/standard/binaries/repo/repoindex.xml](https://build.opensuse.org/projects/openSUSE:Factory/packages/telegram-desktop/repositories/standard/binaries/repo/repoindex.xml)"
### Where ###
binaries#show

### Occured ###
Aug 23  2:49am

### Similar ###
57

## Params ##
~~~
{
  &quot;action&quot; =&gt; &quot;show&quot;,
  &quot;arch&quot; =&gt; &quot;repo&quot;,
  &quot;controller&quot; =&gt; &quot;webui/packages/binaries&quot;,
  &quot;filename&quot; =&gt; &quot;repoindex.xml&quot;,
  &quot;package_name&quot; =&gt; &quot;telegram-desktop&quot;,
  &quot;project_name&quot; =&gt; &quot;openSUSE:Factory&quot;,
  &quot;repository_name&quot; =&gt; &quot;standard&quot;
}
~~~

## Session ##
~~~
{}
~~~

## Backtrace ##
~~~
44:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/journey/formatter.rb -> **path**
824:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/routing/route_set.rb -> **url_for**
271:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/routing/route_set.rb -> **call**
327:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/routing/route_set.rb -> **block in define_url_helper**
30:  app/controllers/webui/packages/main_controller.rb -> **set_architecture**
400:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/callbacks.rb -> **block in make_lambda**
180:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/callbacks.rb -> **block (2 levels) in halting_and_conditional**
34:  /GEM_ROOT/gems/actionpack-7.0.6/lib/abstract_controller/callbacks.rb -> **block (2 levels) in &lt;module:Callbacks&gt;**
181:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/callbacks.rb -> **block in halting_and_conditional**
595:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/callbacks.rb -> **block in invoke_before**
595:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/callbacks.rb -> **each**
595:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/callbacks.rb -> **invoke_before**
106:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/callbacks.rb -> **run_callbacks**
233:  /GEM_ROOT/gems/actionpack-7.0.6/lib/abstract_controller/callbacks.rb -> **process_action**
23:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_controller/metal/rescue.rb -> **process_action**
67:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_controller/metal/instrumentation.rb -> **block in process_action**
206:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/notifications.rb -> **block in instrument**
24:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/notifications/instrumenter.rb -> **instrument**
206:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/notifications.rb -> **instrument**
66:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_controller/metal/instrumentation.rb -> **process_action**
259:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_controller/metal/params_wrapper.rb -> **process_action**
27:  /GEM_ROOT/gems/activerecord-7.0.6/lib/active_record/railties/controller_runtime.rb -> **process_action**
151:  /GEM_ROOT/gems/actionpack-7.0.6/lib/abstract_controller/base.rb -> **process**
39:  /GEM_ROOT/gems/actionview-7.0.6/lib/action_view/rendering.rb -> **process**
188:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_controller/metal.rb -> **dispatch**
251:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_controller/metal.rb -> **dispatch**
49:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/routing/route_set.rb -> **dispatch**
32:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/routing/route_set.rb -> **serve**
18:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/routing/mapper.rb -> **block in &lt;class:Constraints&gt;**
48:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/routing/mapper.rb -> **serve**
50:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/journey/router.rb -> **block in serve**
32:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/journey/router.rb -> **each**
32:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/journey/router.rb -> **serve**
852:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/routing/route_set.rb -> **call**
72:  /GEM_ROOT/gems/flipper-0.28.3/lib/flipper/middleware/memoizer.rb -> **memoized_call**
37:  /GEM_ROOT/gems/flipper-0.28.3/lib/flipper/middleware/memoizer.rb -> **call**
15:  config/initializers/prefer_xml_over_html.rb -> **call**
44:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/deflater.rb -> **call**
15:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/tempfile_reaper.rb -> **call**
27:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/etag.rb -> **call**
27:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/conditional_get.rb -> **call**
12:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/head.rb -> **call**
38:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/http/permissions_policy.rb -> **call**
36:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/http/content_security_policy.rb -> **call**
266:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/session/abstract/id.rb -> **context**
260:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/session/abstract/id.rb -> **call**
704:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/middleware/cookies.rb -> **call**
27:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/middleware/callbacks.rb -> **block in call**
99:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/callbacks.rb -> **run_callbacks**
26:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/middleware/callbacks.rb -> **call**
32:  /GEM_ROOT/gems/airbrake-9.4.3/lib/airbrake/rack/middleware.rb -> **call!**
21:  /GEM_ROOT/gems/airbrake-9.4.3/lib/airbrake/rack/middleware.rb -> **call**
28:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/middleware/debug_exceptions.rb -> **call**
29:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/middleware/show_exceptions.rb -> **call**
18:  /GEM_ROOT/gems/lograge-0.13.0/lib/lograge/rails_ext/rack/logger.rb -> **call_app**
25:  /GEM_ROOT/gems/railties-7.0.6/lib/rails/rack/logger.rb -> **block in call**
99:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/tagged_logging.rb -> **block in tagged**
37:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/tagged_logging.rb -> **tagged**
99:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/tagged_logging.rb -> **tagged**
25:  /GEM_ROOT/gems/railties-7.0.6/lib/rails/rack/logger.rb -> **call**
93:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/middleware/remote_ip.rb -> **call**
93:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/middleware/remote_ip.rb -> **call**
19:  /GEM_ROOT/gems/request_store-1.5.1/lib/request_store/middleware.rb -> **call**
26:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/middleware/request_id.rb -> **call**
24:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/method_override.rb -> **call**
22:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/runtime.rb -> **call**
29:  /GEM_ROOT/gems/activesupport-7.0.6/lib/active_support/cache/strategy/local_cache_middleware.rb -> **call**
14:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/middleware/executor.rb -> **call**
18:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/lock.rb -> **call**
110:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/sendfile.rb -> **call**
131:  /GEM_ROOT/gems/actionpack-7.0.6/lib/action_dispatch/middleware/host_authorization.rb -> **call**
530:  /GEM_ROOT/gems/railties-7.0.6/lib/rails/engine.rb -> **call**
226:  /GEM_ROOT/gems/railties-7.0.6/lib/rails/railtie.rb -> **public_send**
226:  /GEM_ROOT/gems/railties-7.0.6/lib/rails/railtie.rb -> **method_missing**
74:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/urlmap.rb -> **block in call**
58:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/urlmap.rb -> **each**
58:  /GEM_ROOT/gems/rack-2.2.8/lib/rack/urlmap.rb -> **call**
107:  /usr/lib64/ruby/gems/3.1.0/gems/passenger-6.0.14/src/ruby_supportlib/phusion_passenger/rack/thread_handler_extension.rb -> **process_request**
157:  /usr/lib64/ruby/gems/3.1.0/gems/passenger-6.0.14/src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb -> **accept_and_process_next_request**
110:  /usr/lib64/ruby/gems/3.1.0/gems/passenger-6.0.14/src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb -> **main_loop**
419:  /usr/lib64/ruby/gems/3.1.0/gems/passenger-6.0.14/src/ruby_supportlib/phusion_passenger/request_handler.rb -> **block (3 levels) in start_threads**
113:  /usr/lib64/ruby/gems/3.1.0/gems/passenger-6.0.14/src/ruby_supportlib/phusion_passenger/utils.rb -> **block in create_thread_and_abort_on_exception**
~~~

## Environment ##
| Key        | Value      |
|------------|------------|
| program_name | Passenger RubyApp: /srv/www/obs/api (production) |
| HTTP_USER_AGENT | ZYpp 17.31.14 (curl 8.0.1) |

**Comment by danidoni** [comment]
So the offending url is `https://build.opensuse.org/projects/openSUSE:Factory/packages/telegram-desktop/repositories/standard/binaries/x86_86/rpmlint.log` (for example, you could go to `https://build.opensuse.org/projects/openSUSE:Factory/packages/telegram-desktop/repositories/standard/binaries/` and pick another binary file name). That url, tries to build the index action url and creashes. 
This https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/webui/packages/main_controller.rb#L20 needs to read `params[:repository_name]` also, as we can see from the description of this issue.

**Comment by danidoni** [comment]
> Didn't think it affected production, but I did fix that in #13564, in this commit: [5df16af](https://github.com/openSUSE/open-build-service/commit/5df16af9177f8c30709ca1f1f8327f5ee8760a67)

We can fix it in a dedicated PR. Doing so your PR will be smaller.
